otter/convergence/service.py

```diff
@@ -389,7 +394,14 @@ def converge_one_group(currently_converging, tenant_id, group_id, version,
         # unexpected errors, so convergence will be retried.
         yield err(None, 'converge-non-fatal-error')
     else:
-        if result in (StepResult.FAILURE, StepResult.SUCCESS):
+        if result[0] in (StepResult.FAILURE, StepResult.SUCCESS):
+            # In order to avoid doing extra work and reporting spurious errors,
+            # if the group status is DELETING it means the group has
+            # successfully been deleted by execute_convergence. And so we will
+            # unconditionally delete the divergent flag to avoid any further
+            # queued-up convergences that will imminently fail.
+            if result[1] == ScalingGroupStatus.DELETING:
```

Comment:
To answer your first question, it is indeed possible

and thinking about this, I think this whole change is invalid. I realized that we may require multiple convergence iterations to actually delete a group, and this change I made just causes it to delete the divergent flag after the first iteration unconditionally.

We're going to need to distinguish DELETING from "actually deleted".